smartsim/settings/arguments/batch/lsf.py

```diff
@@ -57,8 +57,11 @@ def set_walltime(self, walltime: str) -> None:
         :param walltime: Time in hh:mm format, e.g. "10:00" for 10 hours,
                          if time is supplied in hh:mm:ss format, seconds
                          will be ignored and walltime will be set as ``hh:mm``
+        :raises TypeError: if not type str
         """
         # For compatibility with other launchers, as explained in docstring
+        if not isinstance(walltime, str):
+            raise TypeError("walltime argument was not of type str")
         if walltime:
             if len(walltime.split(":")) > 2:
```

Comment:
For the set function in the lsf module, would you add:

```
if key in self._batch_args and key != self._batch_args[key]:
            logger.warning(f"Overwriting argument '{key}' with value '{value}'")
```

Reply:
added